src/state.rs

```diff
@@ -29,18 +30,19 @@ pub struct State {
 impl State {
     /// Create a new state with the given database_url
     pub async fn new(database_url: &str) -> anyhow::Result<Self> {
-        let db = PgPool::connect(database_url).await?;
+        let db = PgPoolOptions::new()
+            .connect_timeout(std::time::Duration::from_secs(2))
```

Comment:
Maybe it could be a good idea to define the DB timeout limit using an Env Var, what do you think?

Reply:
Done!